### DISCUSSION TITLE
Policy error while converting to image

### DISCUSSION THREAD

**Discussion by kathimohan** [jsvine/pdfplumber#777] [Q&A] [answered]
I tried to convert to image but getting the error - PolicyError: attempt to perform an operation not allowed by the security policy `PDF' @ error/constitute.c/IsCoderAuthorized/421

The code is 

import pdfplumber

pdf_name = "outputk.pdf"
with pdfplumber.open(pdf_name) as pdf:
    pages = len(pdf.pages)
    print(pages)
    p = pdf.pages[0]
    im = p.to_image()

Changed the policy to read/write for pdf  but didn't work

<policy domain="coder" rights="read|write" pattern="PDF" />
  <policy domain="coder" rights="none" pattern="XPS" />
<policy domain="module" rights="read | write" pattern="{GIF,JPEG,PNG,WEBP}" />

![image](https://user-images.githubusercontent.com/36533398/207610897-7547dde7-d214-4bd7-b818-4e36a7304a8a.png)

**Comment by jsvine** [comment]
Hi @kathimohan, do the answers in https://github.com/jsvine/pdfplumber/issues/81#issuecomment-1312277946 help resolve this for you?

**Comment by samkit-jain** [comment, marked as answer]
In file `/etc/ImageMagick-6/policy.xml` (or `/etc/ImageMagick/policy.xml`)

1) comment line
```xml
<!-- <policy domain="coder" rights="none" pattern="MVG" /> -->
```
2) change line
```xml
<policy domain="coder" rights="none" pattern="PDF" />
```
to
```xml
<policy domain="coder" rights="read|write" pattern="PDF" />
```
3) add line
```xml
<policy domain="coder" rights="read|write" pattern="LABEL" />
```

**Then restart your computer.**
